R/truncnorm.R

```diff
@@ -109,6 +114,12 @@ my_vtruncnorm = function(a,b,mean = 0, sd = 1){
   beta =  (b-mean)/sd
   frac1 = (beta*stats::dnorm(beta,0,1) - alpha*stats::dnorm(alpha,0,1)) / (stats::pnorm(beta,0,1)-stats::pnorm(alpha,0,1) )
   frac2 = (stats::dnorm(beta,0,1) - stats::dnorm(alpha,0,1)) / (stats::pnorm(beta,0,1)-stats::pnorm(alpha,0,1) )
+  # ZMZ: deal with NAs when both the numerator and denominator is zero
+  # not = is.na(frac1)
+  # frac1[not]=
   truncnormvar = sd^2 * (1 - frac1 - frac2^2)
+  # turn all nan and negative into 0
```

Comment:
I'm quite sure it is the cause of Nan generated from my_vtruncnorm. The function is short and easy. I tried some numbers and look into the results. It seems to be the reason.